crates/stark-backend/src/keygen/types.rs

```diff
@@ -87,6 +87,20 @@ pub struct MultiStarkVerifyingKey<SC: StarkGenericConfig> {
     pub per_air: Vec<StarkVerifyingKey<Val<SC>, Com<SC>>>,
     pub trace_height_constraints: Vec<LinearConstraint>,
     pub log_up_pow_bits: usize,
+    /// The hash of all other parts of the verifying key. The Fiat-Shamir hasher will
+    /// initialize by observing this hash.
+    pub pre_hash: Com<SC>,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(bound(
+    serialize = "Com<SC>: Serialize",
+    deserialize = "Com<SC>: Deserialize<'de>"
+))]
+pub(crate) struct MultiStarkVerifyingKey0<SC: StarkGenericConfig> {
```

Comment:
This means every time we want to add a field to the verifying key we need to remember to add it here too, right? Can we add a const assert that `size_of(MultiStarkVerifyingKey0) == size_of(MultiStarkVerifyingKey) - size_of(Com)` or something?

Reply:
interesting

Reply:
the generic `SC` was a blocker

Reply:
switched to `inner` pattern